great_tables/_formats.py

```diff
@@ -285,66 +286,129 @@ def fmt_number(
+                fmt_number_context,
+                data=self,
+                decimals=decimals,
+                n_sigfig=n_sigfig,
+                drop_trailing_zeros=drop_trailing_zeros,
+                drop_trailing_dec_mark=drop_trailing_dec_mark,
+                use_seps=use_seps,
+                scale_by=scale_by,
+                compact=compact,
+                sep_mark=sep_mark,
+                dec_mark=dec_mark,
+                force_sign=force_sign,
+                pattern=pattern,
+                context="html",
+            ),
+        ),
+        columns=columns,
+        rows=rows,
+    )
 
-        # Implement minus sign replacement for `x_formatted`
-        if is_negative:
-            minus_mark = _context_minus_mark()
-            x_formatted = _replace_minus(x_formatted, minus_mark=minus_mark)
 
-        # Use a supplied pattern specification to decorate the formatted value
-        if pattern != "{x}":
-            x_formatted = pattern.replace("{x}", x_formatted)
+def fmt_number_context(
+    x: float | None,
+    data: GTData,
+    decimals: int,
+    n_sigfig: int | None,
+    drop_trailing_zeros: bool,
+    drop_trailing_dec_mark: bool,
+    use_seps: bool,
+    scale_by: float,
+    compact: bool,
+    sep_mark: str,
+    dec_mark: str,
+    force_sign: bool,
+    pattern: str,
+    context: str,
+):
+    if is_na(data._tbl_data, x):
+        return x
 
-        return x_formatted
+    # Scale `x` value by a defined `scale_by` value
+    x = x * scale_by
+
+    # Determine whether the value is positive
+    is_negative = _has_negative_value(value=x)
 
-    return fmt(self, fns=fmt_number_fn, columns=columns, rows=rows)
+    if compact:
+        x_formatted = _format_number_compactly(
+            value=x,
+            decimals=decimals,
+            n_sigfig=n_sigfig,
+            drop_trailing_zeros=drop_trailing_zeros,
+            drop_trailing_dec_mark=drop_trailing_dec_mark,
+            use_seps=use_seps,
+            sep_mark=sep_mark,
+            dec_mark=dec_mark,
+            force_sign=force_sign,
+        )
+    else:
+        x_formatted = _value_to_decimal_notation(
+            value=x,
+            decimals=decimals,
+            n_sigfig=n_sigfig,
+            drop_trailing_zeros=drop_trailing_zeros,
+            drop_trailing_dec_mark=drop_trailing_dec_mark,
+            use_seps=use_seps,
+            sep_mark=sep_mark,
+            dec_mark=dec_mark,
+            force_sign=force_sign,
+        )
+
+    # Implement minus sign replacement for `x_formatted`
+    if is_negative:
+        minus_mark = _context_minus_mark(context=context)
+        x_formatted = _replace_minus(x_formatted, minus_mark=minus_mark)
+
+    # Use a supplied pattern specification to decorate the formatted value
+    if pattern != "{x}":
+
+        # Escape LaTeX special characters from literals in the pattern
+        if context == "latex":
+            pattern = escape_pattern_str_latex(pattern_str=pattern)
```

Comment:
The function `escape_pattern_str_latex()` is used throughout the formatting functions here to modify the `pattern` string for the `"latex"` context only (for this PR). These three lines will be included in every `if pattern != "{x}":` conditional below.